### ISSUE TITLE
tslint should throw an error when rules are in an unexpected format (specifically "error" instead of true)

### ISSUE THREAD

**Issue by pixelpax** [palantir/tslint#3024]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 0.24.5
- __Running TSLint via__: (pick one) CLI and WebStorm

#### TypeScript code being linted

```ts
import * as React from 'react';

export class Josh {
  constructor(public x: number) {
  }

  public sayX(dog: number[]) 
    let f: string = 'dogs';
      return this.x;
  }
}```

with `tslint.json` configuration:

```json
{
  "extends": ["tslint:latest", "tslint-react", "tslint-eslint-rules"],
  "rules": {
    // ts specific
    "array-type": ["error", "array"],
    "arrow-parens": ["error"],

    // js general
    "class-name": "error",
    "variable-name": "error",
    "no-empty": ["error", "allow-empty-catch"],
    "indent": ["error", "spaces", 2],                       // Two spaces per indentation, no tabs allowed (fixable)
    "semicolon": "error",                                   // Enforces semicolons after statements (fixable)
    "whitespace": [
      "error",
      "check-branch",
      "check-operator",
      "check-decl",
      "check-separator",
      "check-type",
      "check-typecast",
      "check-preblock",
      "check-module"
    ],
    "no-trailing-whitespace": "error",                      // Enforces trailing whitespace (fixable)
    "trailing-comma": ["error", "never"],                   // No trailing commmas (fixable)
    "no-multi-spaces": ["error", {}],                       // One space per gap (fixable)
    "eofline": "error",                                     // Mandate newlines at the end of files (fixable)
    "quotemark": ["error", "single", "jsx-single", "avoid-escape"],
    "no-consecutive-blank-lines": "error",                  // No overuse of newlines (fixable)
    "triple-equals": "error",
    "comment-format": ["error", "check-space"],             // Put a space after inline comments (fixable)
    "no-debugger": "error",                                 // Don't allow submissions which retain debugger statements
    "no-arg": "error"                                       // No access of `caller` and `callee` attributes. Performance killers
  }
}
```

#### Actual behavior
Coming from ESLint myself (and many others I'm sure) have .eslint files which turn on rules using `error` and `warn`. This method is preferred for many, because it makes eslint files immediately understandable to team members who are not familiar w/ the syntax. 

When used in tslint, no error is thrown, but the rule will not be turned on. 

#### Expected behavior
An error explicitly telling me that strings are not a valid type in this slot. Would've saved me an hour of trying to figure out what was wrong with my setup.

**Comment by JoshuaKGoldberg** [comment]
Agreed. Duplicate of #889.
